Compilation Error with digitalPinCanOutput(pin) After Update from 2.0.14 to 3.0.3

### Board

OLIMEX ESP32-PoE-ISO

### Device Description

irrelevant

### Hardware Configuration

irrelevant

### Version

v3.0.3

### IDE Name

Arduino IDE 2.3.2

### Operating System

Windows 10

### Flash frequency

80Mhz

### PSRAM enabled

no

### Upload speed

921600

### Description

Hello,

After updating from version 2.0.14 to 3.0.3 of the Arduino-ESP32 library, I encountered a compilation error when using the function digitalPinCanOutput(pin).

The error message is as follows:

```compiler

...\Arduino15\packages\esp32\hardware\esp32\3.0.3\cores\esp32/esp32-hal-gpio.h:68:34: error: 'GPIO_IS_VALID_OUTPUT_GPIO' was not declared in this scope
   68 | #define digitalPinCanOutput(pin) GPIO_IS_VALID_OUTPUT_GPIO(pin)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~
```
It appears that GPIO_IS_VALID_OUTPUT_GPIO is not declared in the scope, which causes the compilation to fail.

### Sketch

```cpp
bool canOutput = digitalPinCanOutput(0);
```


### Debug Message

```plain
irrelevant
```


### Other Steps to Reproduce

It still worked with version 2.0.14.

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

### Edit
I was able to get around the problem by including "driver/gpio.h" in the first line of my sketch.

--- a/cores/esp32/esp32-hal-gpio.h
+++ b/cores/esp32/esp32-hal-gpio.h
@@ -27,6 +27,7 @@ extern "C" {
 #include "esp32-hal.h"
 #include "soc/soc_caps.h"
 #include "pins_arduino.h"
+#include "driver/gpio.h"
 
 #if (CONFIG_IDF_TARGET_ESP32S2 || CONFIG_IDF_TARGET_ESP32S3)
 #define NUM_OUPUT_PINS 46

